# HG changeset patch
# User Rik <rik@octave.org>
# Date 1355094444 28800
#      Sun Dec 09 15:07:24 2012 -0800
# Node ID 9f3656ada65814ca85cf95c622d8812392864998
# Parent  1ac3c906990a45169b74e08245f70db7d745aefb
configure.ac: Add warning message if JAVA_HOME variable is not set.

* configure.ac: Add warning message if JAVA_HOME variable is not set.

diff --git a/configure.ac b/configure.ac
--- a/configure.ac
+++ b/configure.ac
@@ -2215,28 +2215,33 @@ AC_ARG_ENABLE([java],
     [disable Java interface])],
   [if test "$enableval" = no; then
      build_java=false
    fi],
   [])
 
 HAVE_JAVA=no
 ## Grab JAVA_HOME from environment variable if it exists
-AC_ARG_VAR([JAVA_HOME], [path to Java installation])
+AC_ARG_VAR([JAVA_HOME], [path to Java JDK installation])
 JAVA_ARCH=
 JAVAC=
 JAR=
 JAVA_CPPFLAGS=
 JAVA_LIBS=
 
 if test $build_java = true; then
 
   warn_java=
 
   if test -z "$JAVA_HOME"; then
+    AC_MSG_WARN([JAVA_HOME environment variable not initialized.])
+    AC_MSG_WARN([Auto-detection will proceed but is unreliable.])
+  fi
+
+  if test -z "$JAVA_HOME"; then
     AC_CHECK_PROG(JAVA, java, java)
     AC_CHECK_PROG(JAVAC, javac, javac)
     AC_CHECK_PROG(JAR, jar, jar)
   else
     AC_PATH_PROG(JAVA, java, [], [$JAVA_HOME/bin$PATH_SEPARATOR$PATH])
     AC_PATH_PROG(JAVAC, javac, [], [$JAVA_HOME/bin$PATH_SEPARATOR$PATH])
     AC_PATH_PROG(JAR, jar, [], [$JAVA_HOME/bin$PATH_SEPARATOR$PATH])
   fi
